src/BackupManager/Factory/StorageFactory.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace DigipolisGent\Robo\Task\Deploy\BackupManager\Factory;
+
+use BackupManager\Config\Config;
+use BackupManager\Filesystems\Awss3Filesystem;
+use BackupManager\Filesystems\DropboxFilesystem;
+use BackupManager\Filesystems\FilesystemProvider;
+use BackupManager\Filesystems\FtpFilesystem;
+use BackupManager\Filesystems\GcsFilesystem;
+use BackupManager\Filesystems\LocalFilesystem;
+use BackupManager\Filesystems\RackspaceFilesystem;
+use BackupManager\Filesystems\SftpFilesystem;
+
+class StorageFactory implements StorageFactoryInterface
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function create($storageConfig)
+    {
+        $storageConfigObj = is_array($storageConfig)
+            ? new Config($storageConfig)
+            : Config::fromPhpFile($storageConfig);
+
+        $filesystemProvider = new FilesystemProvider($storageConfigObj);
```

Comment:
StorageProvider or FilesystemProvider? We should choose one and name the ProviderFactory and Provider the same way.